lms/djangoapps/mobile_api/course_info/tests.py

```diff
@@ -84,60 +84,95 @@ def test_updates(self, new_format):
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_handouts(self, default_ms):
+        with self.store.default_store(default_ms):
+            self.add_mobile_available_toy_course()
+            response = self.api_response(expected_response_code=200)
+            self.assertIn("Sample", response.data['handouts_html'])
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_no_handouts(self, default_ms):
+        with self.store.default_store(default_ms):
+            self.add_mobile_available_toy_course()
+
+            # delete handouts in course
+            handouts_usage_key = self.course.id.make_usage_key('course_info', 'handouts')
+            with self.store.branch_setting(ModuleStoreEnum.Branch.draft_preferred, self.course.id):
+                self.store.delete_item(handouts_usage_key, self.user.id)
+
+            response = self.api_response(expected_response_code=200)
+            self.assertIsNone(response.data['handouts_html'])
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_empty_handouts(self, default_ms):
+        with self.store.default_store(default_ms):
+            self.add_mobile_available_toy_course()
+
+            # set handouts to empty tags
+            handouts_usage_key = self.course.id.make_usage_key('course_info', 'handouts')
+            underlying_handouts = self.store.get_item(handouts_usage_key)
+            underlying_handouts.data = "<ol></ol>"
+            self.store.update_item(underlying_handouts, self.user.id)
+            response = self.api_response(expected_response_code=200)
+            self.assertIsNone(response.data['handouts_html'])
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_handouts_static_rewrites(self, default_ms):
+        with self.store.default_store(default_ms):
+            self.add_mobile_available_toy_course()
+
+            # check that we start with relative static assets
+            handouts_usage_key = self.course.id.make_usage_key('course_info', 'handouts')
+            underlying_handouts = self.store.get_item(handouts_usage_key)
+            self.assertIn('\'/static/', underlying_handouts.data)
+
+            # but shouldn't finish with any
+            response = self.api_response()
+            self.assertNotIn('\'/static/', response.data['handouts_html'])
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_jump_to_id_handout_href(self, default_ms):
+        with self.store.default_store(default_ms):
+            self.add_mobile_available_toy_course()
+
+            # check that we start with relative static assets
+            handouts_usage_key = self.course.id.make_usage_key('course_info', 'handouts')
+            underlying_handouts = self.store.get_item(handouts_usage_key)
+            underlying_handouts.data = "<a href=\"/jump_to_id/identifier\">Intracourse Link</a>"
```

Comment:
This is the style of URL we allow instructors to use when they are creating courses. For example, MIT's "Minds and Machines" course uses:

<ol>
  <a href="/jump_to_id/2b02305f3c494aa0b7f0d43aa0778fa9">Discussion Forum Guidelines</a>
</ol>
<ol>
  <a href="/jump_to_id/095f84e31ecf485bb14b51e7b7991db8">Class Schedule</a>
</ol>

https://studio.edx.org/course_info/course-v1:MITx+24.09x+3T2015#